src/shared/slideRegistry.ts

```diff
@@ -0,0 +1,92 @@
+import { ComponentInternalInstance, shallowReactive } from 'vue'
+
+type SlideEventName = 'register' | 'unregister'
+
+interface SlideRegistrationEvent {
+  slide: ComponentInternalInstance
+  index: number
+}
+
+type SlideRegistrationCallback = (event: SlideRegistrationEvent) => void
+
+const createSlideRegistry = () => {
+  const slides = shallowReactive<Array<ComponentInternalInstance>>([])
+  const listeners: Record<SlideEventName, Set<SlideRegistrationCallback>> = {
+    register: new Set(),
+    unregister: new Set(),
+  }
+
+  const updateSlideIndexes = (startIndex?: number) => {
+    if (startIndex !== undefined) {
+      // Update only slides from startIndex onwards
+      slides.slice(startIndex).forEach((slide, offset) => {
+        slide.props.index = startIndex + offset
```

Comment:
Maybe using expose of a ref here in the slide and removing the index prop here rather than mutating props is best

Reply:
While mutating the index prop from the parent component should not be harmful in this context, I understand the concerns about VNode prop reactivity and potential VNode API misuse.

I agree that using expose with a ref for the index provides a cleaner, more Vue-idiomatic solution. However, I need to rethink this whole approach to handle cases where slides are dynamically added in the middle of the sequence.